h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_add_te_model_to_a_regular_model.py

```diff
@@ -25,24 +26,118 @@ def test_target_encoder_model_summary_does_not_contain_fold_column():
     print(te)
     print(trainingFrame)
 
-    model_summary = te._model_json['output']['model_summary'].as_data_frame()
-    print(model_summary)
-    encoded_column_names = model_summary['encoded_column_name']
+    myX = ["pclass", "sex", "age", "sibsp", "parch", "fare", "cabin", "embarked", "home.dest"]
+    titanic_gbm_model = H2OGradientBoostingEstimator(ntrees=50,
+                                             learn_rate=0.1,
+                                             score_tree_interval=10,
+                                             stopping_rounds=5,
+                                             stopping_metric="AUC",
+                                             stopping_tolerance=0.001,
+                                             distribution="multinomial",
+                                             categorical_encoding="enum_limited",
+                                             te_model_id = te,
```

Comment:
@sebhrusen @ledell got a question. On a backend we expect `te_model_id` to be a key ( string), that is why autogenerated documentation in Py says type should be `str`:

```
 @property
    def te_model_id(self):
        """
        Key of TargetEncoderModel

        Type: ``str``.
        """
```
On the other hand, for validation against type of model (its class) on Python side we need to pass Model, not id. This might be confusing for users ( not sure if we have convention that whenever id of model is required we can pass model itself). 
 
So below, we do check of type even though docs say it could be only string.
```
assert_is_type(te_model_id, None, H2OTargetEncoderEstimator)
```
The only place (that I found) where we accept model itself and do assertion against class is in `grid_search.py` - but this file is not autogenerated one and docs do not contradict with code.

Probably we should not validate against class or otherwise we can try to change somehow gen_python.py logic. With `str` type we will not be able to validate on client side (only by checking suffix but not sure if we want to do it like this).

Reply:
@sebhrusen I added a hook for doc as well but it appends not overrides. We need overrides for docs as well.

Reply:
@sebhrusen figured this out. It does not append.... there is also a help section that comes before docs.

Reply:
You're talking about getters or setters here?
Your `te_model_id` will always be a `str`, right?
Currently, for some unknown reason, there's no docstring generated for setters, but you can add one if you need it:
in `gen_python.py`,
just change 
``` 
custom_property_doc = get_customizations_for(algo, "doc.{}".format(pname))
```
to
```
        custom_property_doc = get_customizations_or_defaults_for(algo, "doc.getter.{}".format(pname))
```
and a few lines below, you can add a similar hook for setter.

Reply:
actually it may not even be necessary, you can just add the doc string in your existing setter hook.

Reply:
also, you were complaining about code being represented as strings.
Yes, that's how the code generator works and has always been working.
But for Python, it's a bit more clever now, this should work:
```
def te_model_id_setter():
    """ your docstring"""
    your_python_code

overrides=dict(
    te_model_id=dict(
          setter=te_model_id_setter
    )
)

```

Reply:
no, this shouldn't work out of the box, but it should if you call `code_as_str` :
```
property_getter = code_as_str(get_customizations_for(algo, "overrides.{}.getter".format(pname)))
```
same for setter

Reply:
You could follow whatever conventions we have already established with frames.  For all `training_frame` etc arguments, you can pass either an H2OFrame or an ID... so it's the same idea here.

Reply:
@ledell ok, as I thought. Will rename `te_model_id` to `te_model` then.

Reply:
@sebhrusen 
> also, you were complaining about code being represented as strings.
Yes, that's how the code generator works and has always been working.

sure, but mostly we generate code dynamically and this hook is fixed/hardcoded and also it is not super easy to find from `gen_python.py`.  But anyway, that's fine.

> You're talking about getters or setters here?

I was talking about setter. If we allow only strings ( keys of models) then there is probably no need for hooks at all. But I will try to keep ability to pass model itself.